library/alloc/src/gc.rs

```diff
@@ -98,6 +98,36 @@ impl<T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<Gc<U>> for Gc<T> {}
 #[unstable(feature = "gc", issue = "none")]
 impl<T: ?Sized + Unsize<U>, U: ?Sized> DispatchFromDyn<Gc<U>> for Gc<T> {}
 
+/// A memory barrier to prevent finalizers running before the last reference to
```

Comment:
"Barriers" are both a compile-time and a run-time concept. This is a "compiler" barrier, right?

Reply:
That's right. Do you think it'd be clearer if I changed this from memory barrier to compiler barrier?

Reply:
I do indeed :)